sklearn/calibration.py

```diff
@@ -478,6 +467,33 @@ def predict(self, X):
         check_is_fitted(self)
         return self.classes_[np.argmax(self.predict_proba(X), axis=1)]
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self(self)
+            .add(
+                estimator=self._get_estimator(),
+                method_mapping=MethodMapping().add(callee="fit", caller="fit"),
+            )
```

Comment:
good catch! yes we do. we should add

``` python
.add(splitter=self._get_cv(), method_mapping=MethodMapping().add(callee="split", caller="fit"))
```

and handle splitter args in fit the same way as for `estimator`.

Reply:
One problem here is that `self._get_cv` would need to get `y` as argument, because `check_cv` takes `y` (and for other stuff). However, that means we cannot call `self._get_cv` in `get_metadata_routing`, since that method does not receive `y`.  

I'm not exactly sure what happens if we just pass `.add(splitter=self.cv, ...)`. It looks like we get a `MetadataRequest(owner=None)` if `self.cv` is, say, an int, is that insufficient?

Tangent: The name "splitter" is arbitrary, right? We could call it anything, as long as we later use the same name when referencing the metadata.

Reply:
You're right. I suppose in this case we could actually just pass `self.cv` since the only thing which matters is the metadata requests, and as long as the user doesn't pass a Group*CV there are not requested metadata.

>Tangent: The name "splitter" is arbitrary, right? We could call it anything, as long as we later use the same name when referencing the metadata.

Correct!

Reply:
Done.